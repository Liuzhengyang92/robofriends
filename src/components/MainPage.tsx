import React from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';
import Header from '../components/Header';
import './MainPage.css';

class MainPage extends React.Component<any> {
  componentDidMount() {
    console.log(this.props);
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter((robot: any) => {
      return robot.name.toLocaleLowerCase().includes(this.props.searchField.toLocaleLowerCase());
    });
  }

  render() {
    const { onSearchChange, isPending } = this.props;
    
    return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? <h1>Loading...</h1> :
            <ErrorBoundary>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundary>
          }
        </Scroll>
      </div>
    );
  }
};

export default MainPage;