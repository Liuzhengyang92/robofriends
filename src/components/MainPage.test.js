import React from 'react';
import { shallow, mount, render } from 'enzyme';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  };
  wrapper = shallow(<MainPage { ...mockProps } />);
});
// it('expect to render App component', () => {
//   const mockStore = {
//     robots: [],
//     searchField: ''
//   };
//   expect(shallow(<MainPage store={mockStore} />)).toMatchSnapshot();
// });

it('renders MainPage without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

it('filters robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'john',
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual([{
    id: 3,
    name: 'John',
    email: 'john@gmail.com'
  }]);
  // expect(wrapper2.instance().filterRobots()).toEqual([]);
  // expect(wrapper.instance().filterRobots(mockProps.robots))
});


it('filters robots correctly2', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'a',
    isPending: false
  };
  const filterRobots = [];
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.instance().filterRobots()).toEqual(filterRobots);
});

const mockProps4 = {
  onRequestRobots: jest.fn(),
  robots: [],
  searchField: '',
  isPending: true
};
const wrapper4= shallow(<MainPage {...mockProps4} />);

it('test isPending is true', () => {
  expect(wrapper4).toMatchSnapshot();
});