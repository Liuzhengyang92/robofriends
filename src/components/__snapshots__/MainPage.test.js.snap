// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`test isPending is true 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <h1>
      Loading...
    </h1>
  </Scroll>
</div>
`;
